### PR TITLE
Feature/gh75 documentation for unittest raft

### LINKED ISSUE #75: Documentation for the unittest implementation of raft
## Goal
Document usage of raft test case/suite.

## Acceptance Criteria
- Getting started guide is created for using raft unitttest 
- Example test is created (similar to example_test.py)

### PR DESCRIPTION
**Pull request by TB-1993**
Added documented example test for new unittest style of writing tests.
Fixed a bug in SSHConsole using port incorrectly.

closes #75

### CODE REVIEW

File: examples/code/example_ssh_unittest.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    def setUp(self):
+        """
+        Perform pre-test setup tasks.
+
+        This method will run before each test method.
+        """
+        self.log.info('Pre-test check')
+        self.log.info('Make the test directory')
+        # Get the default console session for the DUT
+        self.session = self.dut.getConsoleSession()
```
Comment (Ulrond) [inline examples/code/example_ssh_unittest.py:59]: self.session should already exist surely, the framework should have done that?
self.power
self.device
self.log
etc..
should all be setup.

File: examples/code/example_ssh_unittest.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            self.log.info('Rerunning pre-test check to ensure file has been removed')
+
+    def test_fileCreation(self):
+        """
+        The main test method that verifies file creation on the DUT
+
+        Any method starting with the word test at the start of it name,
+        will run as an individual test method.
+        """
+        self.log.stepStart(f'Creating {TEST_FILE} in {TEST_DIRECTORY}')
```
Comment (Ulrond) [inline examples/code/example_ssh_unittest.py:82]: do we still need stepStart, isn't that handled in the framework?

**Review by Ulrond — CHANGES_REQUESTED** [review]
session seems weird. Where's the code commit review for the new framework?

File: examples/code/example_ssh_unittest.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        self.log.info('Make the test directory')
+        # Ping Test to check Box alive
+        if self.dut.pingTest() is False:
+            raise ConnectionError('Cannot reach dut')
+        # Open the console session
+        self.dut.session.open()
+        # Create test directory if if doesn't already exist
+        self.dut.session.write(f'mkdir -p {TEST_DIRECTORY}')
+        # List the files in the test directory
+        self.dut.session.write(f'ls {TEST_DIRECTORY}')
```
Comment (zghp) [inline examples/code/example_ssh_unittest.py:66]: self.dut.session.write(f'ls {TEST_DIRECTORY}')
        file_list = self.dut.session.read_all()

the code is repeated twice in both methods so maybe can organise this bit into a helper method (e.g. list_files())

**Review by zghp — CHANGES_REQUESTED** [review]

File: examples/code/example_ssh_unittest.py [hunk 4]
```diff
@@ -47,6 +49,24 @@ class TestExampleSSH(RAFTUnitTestCase):
...
+
+        Args:
+            directory (str): Path of directory to list files in.
+
+        Returns:
+            str: Space separated list of files from the directory.
+        """
+        # List the files in the test directory
+        self.dut.session.write(f'ls {directory}')
+        file_list = utilities.strip_ansi_escapes(self.dut.session.read_all())
```
Comment (Ulrond) [inline examples/code/example_ssh_unittest.py:64]: These are bad examples, and I had to fix the usage of it for the other teams.

we need to ensure we wait for a prompt when we run commands in all cases for all things.

The waitForSessionMessagePromp() needs to return the text of what was waited for, and it needs to wait for a prompt.

Comment (Ulrond) [reply examples/code/example_ssh_unittest.py:64]: `session.read_all()` must never be used in any of our examples.

Comment (Ulrond) [reply examples/code/example_ssh_unittest.py:64]: or this -> `writeCommandOnDevice(self, command, prompt=None, device="dut"):` -> https://github.com/rdkcentral/ut-raft/pull/4/files#diff-f4c4730e180ac270dfa2f6cd02717b1ae94a0d105c2e09bcde08eebbd5b3c89dR270

**Pull request merged**